### ISSUE TITLE
Proposal: NumberBox V2 - Polish API with Additional Properties and Events

### ISSUE THREAD

**Issue by SavoySchuler** [microsoft/microsoft-ui-xaml#3158]
# Proposal: NumberBox V2 - Polish API with Additional Properties and Events

## Summary

Center NumberBox V2 around polishing API surface by exposing or adding customer requested APIs for additional properties and events. 

## Rationale

Larger NumberBox features are still blocked on requiring WinUI 3.0, however, a meaningful V2 can still be delivered to active customers by focusing on exposing or adding customer requested properties and events. 

See linked issues for each APIs individual rationale/justification. 

## Scope

Requests to expose APIs:
 
1.	Expose InputScope API:
 a.	Proposal https://github.com/microsoft/microsoft-ui-xaml/issues/2710
 b.	PR: https://github.com/microsoft/microsoft-ui-xaml/pull/2605
2.	Styling APIs: https://github.com/microsoft/microsoft-ui-xaml/issues/2844
a.	Border Properties: https://github.com/microsoft/microsoft-ui-xaml/issues/1925
b.	NumberBox should respect Background/Foreground properties: https://github.com/microsoft/microsoft-ui-xaml/issues/2998
c.	Provide a PlaceholderForeground API: https://github.com/microsoft/microsoft-ui-xaml/issues/2811
i.	See also - Controls providing a PlaceholderText API should also provide a PlaceholderForeground API:https://github.com/microsoft/microsoft-ui-xaml/issues/2965
3.	SpinButtons Tooltips: https://github.com/microsoft/microsoft-ui-xaml/issues/2808
 

Requests to add properties, events, theme resources:
 
1.	[Property] Add IsReadOnly / IsValueLocked property: https://github.com/microsoft/microsoft-ui-xaml/issues/2204
2.	[Events] Add Paste, CopyingToClipboard, and CuttingToClipboard Events: https://github.com/microsoft/microsoft-ui-xaml/issues/2055
3.	[Property] Add 'DeleteButtonVisibility' property: https://github.com/microsoft/microsoft-ui-xaml/issues/2205
4.	[Theme Resources] Styling resources for glyphs: https://github.com/microsoft/microsoft-ui-xaml/issues/2789

**Comment by robloo** [comment]
I will circle back around to number 1 (#2710) hopefully this weekend and open a PR.

**Comment by Poopooracoocoo** [comment]
https://github.com/microsoft/Xaml-Controls-Gallery/issues/534

**Comment by myokeeh** [comment]
When is `IsReadOnly` going to be available for `NumberBox`?

**Comment by robloo** [comment]
@myokeeh Might be best to talk about that in the other issue https://github.com/microsoft/microsoft-ui-xaml/issues/2204. I've heard zero movement on any updates to NumberBox. Microsoft is putting all their resources into WinUI 3 and only working on must-haves. All supplemental development like this seems to be on hold.

I added the InputScope property and that PR ended up taking months to get merged (when it should have been days). Needless to say I don't have a lot of motivation to add an IsReadOnly property right now. I'm not even sure I'll be using WinUI a year from now based on how things are progressing.

**Comment by myokeeh** [comment]
@SavoySchuler, this an example of the sentiment I have that I expressed in the last community call that it seems everything we need now is always a year away. #2204 is from April 1, 2020. We're literally a week away from 1 year.
